lib/amazon_sns_helper.rb

```diff
@@ -124,9 +124,11 @@ def self.publish_android(user, target_arn, payload)
     end
   end
 
-  def self.generate_message(payload)
+  def self.generate_message(payload, user)
     message = "@#{payload[:username]}: #{payload[:excerpt]}"
-    message = DiscoursePluginRegistry.apply_modifier(:amazon_sns_message, message, payload)
+    I18n.with_locale(user.effective_locale) do
+      message = DiscoursePluginRegistry.apply_modifier(:amazon_sns_message, message, payload)
+    end
     message
```

Comment:
I think this can be golfed:

```suggestion
    I18n.with_locale(user.effective_locale) do
      DiscoursePluginRegistry.apply_modifier(:amazon_sns_message, message, payload)
    end
```